boot/freeldr/freeldr/lib/peloader.c

```diff
@@ -624,12 +624,16 @@ PeLdrAllocateDataTableEntry(
         FrLdrHeapFree(DataTableEntry, TAG_WLDR_DTE);
         return FALSE;
     }
-    RtlZeroMemory(Buffer, Length);
 
     DataTableEntry->BaseDllName.Length = Length;
     DataTableEntry->BaseDllName.MaximumLength = Length;
     DataTableEntry->BaseDllName.Buffer = PaToVa(Buffer);
-    while (*BaseDllName != 0)
+
+    // Save Buffer, in case of later failure.
+    BaseDllNameBuffer = Buffer;
```

Comment:
might be moved just after the FrLdrHeapAlloc + NULL-check above.

Reply:
If moved, that could be up to the allocation itself, for simplicity.
However, I would prefer to have it here, to emphasize that `Buffer` will change after that point.